File: backend/Models/Referral.cs
namespace CZConnect.Models;

public class Referral 
{
    public long Id { get; set; }
    public string ParticipantName { get; set; }

    public ReferralStatus Status { get; set; }
    public string? ParticipantEmail { get; set; }
    public string? ParticipantPhoneNumber { get; set; }
    public DateTime RegistrationDate { get; set; }
    public long EmployeeId { get; set; }
    public Employee? Employee { get; set; }
}

File: cz_app/lib/widget/app/Dashboard/models/Referral.dart
class Referral {
  int id;
  String status;
  String participantName;
  String participantEmail;
  DateTime registrationDate;

  Referral(
      {required this.id,
      required this.status,
      required this.participantName,
      required this.participantEmail,
      required this.registrationDate});

  factory Referral.fromJson(Map<String, dynamic> json) {
    return Referral(
      id: json['id'],
      participantName: json['participantName'],
      participantEmail: json['participantEmail'],
      status: json['status'],
      registrationDate: DateTime.parse(json['registrationDate']),
    );
  }
}
